src/compiler/llvm/mir.rs

```diff
@@ -596,13 +609,33 @@ impl<'p, 'module, 'ctx> FunctionBuilder<Location<'ctx>, BasicValueEnum<'ctx>>
         }
     }
 
+    fn alloc_arg(&mut self, arg_id: ArgId, decl: &ArgDecl) -> Result<(), TransformerError> {
+        let name = self.arg_label(decl);
+        let arg_value = self.get_arg(arg_id)?;
+        arg_value.set_name(&name);
+
+        // Check if variable name already exists
+        let var_id = decl.var_id().unwrap();
+        match self.vars.entry(var_id) {
+            // If it does -> return an error
+            Entry::Occupied(_) => Err(TransformerError::VariableAlreadyAllocated),
+
+            // If not, then allocate a pointer in the Builder
+            Entry::Vacant(ve) => {
+                // and add a mapping from VarID to the pointer in the local var table
+                ve.insert(Location::Argument(arg_value));
+                Ok(())
+            }
+        }
+    }
+
     fn alloc_var(&mut self, id: VarId, decl: &VarDecl) -> Result<(), TransformerError> {
         let name = self.var_label(decl);
 
         // Check if variable name already exists
         match self.vars.entry(id) {
             // If it does -> return an error
-            Entry::Occupied(_) => Err(TransformerError::VariableAlreadyAllocated),
+            Entry::Occupied(_) => Ok(()),
```

Comment:
Check to make sure this is allocated with an argument, if not then throw an error.

The only way a variable should already be inserted into the hashmap is when it mirrored an argument.